suite-common/firmware/src/hooks/useFirmwareInstallation.ts

```diff
@@ -227,8 +234,15 @@ export const useFirmwareInstallation = (
     }, [originalDevice, shouldSwitchFirmwareType]);
 
     const firmwareUpdate = useCallback(
-        (arg: FirmwareUpdateProps) => dispatch(firmwareUpdateThunk(arg)),
-        [dispatch],
+        (arg: FirmwareUpdateProps) => {
+            if (isDeviceConnectedViaBluetoothLowOnBattery) {
+                dispatch(firmwareActions.setFirmwareUpdateOnLowBattery(true));
```

Comment:
I rather imperatively switch the flag for the UI on, instead of just reacting to the update / firmware state.

It looks to me transparent, but?